contracts/mainnet/connectors/notional/SafeInt256.sol

```diff
@@ -0,0 +1,31 @@
+pragma solidity ^0.7.6;
+
+library SafeInt256 {
+    int256 private constant _INT256_MIN = type(int256).min;
+
+    function mul(int256 a, int256 b) internal pure returns (int256 c) {
+        c = a * b;
+        if (a == -1) require(b == 0 || c / b == a);
+        else require(a == 0 || c / a == b);
+    }
+
+    function div(int256 a, int256 b) internal pure returns (int256 c) {
+        require(!(b == -1 && a == _INT256_MIN)); // dev: int256 div overflow
+        // NOTE: solidity will automatically revert on divide by zero
+        c = a / b;
+    }
+
+    function sub(int256 x, int256 y) internal pure returns (int256 z) {
+        //  taken from uniswap v3
+        require((z = x - y) <= x == (y >= 0));
+    }
+
+    function neg(int256 x) internal pure returns (int256 y) {
+        return mul(-1, x);
+    }
+
+    function toInt(uint256 x) internal pure returns (int256 y) {
+        require(x <= uint256(type(int256).max));
+        return int256(x);
+    }
```

Comment:
I think we don't have to use this safe library, and we do have the same library in `~/contracts/mainnet/common/math.sol`, which has all this functionality.
If something in it is missing, you can add it in there.

Reply:
I removed this file and added toUint to DSMath